runtime/vm/resolvesupport.cpp

```diff
@@ -870,8 +869,7 @@ resolveInstanceFieldRefInto(J9VMThread *vmStruct, J9Method *method, J9ConstantPo
 			}
 
 			if ((resolveFlags & J9_RESOLVE_FLAG_FIELD_SETTER) != 0 && (modifiers & J9_JAVA_FINAL) != 0) {
-				checkResult = checkVisibility(vmStruct, classFromCP, definingClass, J9_JAVA_PRIVATE, resolveFlags);
-				if (checkResult < J9_VISIBILITY_ALLOWED) {
+				if ((classFromCP == definingClass) || (J9ROMCLASS_IS_UNSAFE(classFromCP->romClass) != 0)) {
```

Comment:
This looks like it flips the check from saying: `If checkVisibility says no, fail` to `if checkVisible would have said yes, fail`.

How has this been tested?

Reply:
Accidentally pushed the wrong commit; fixed now.

I've been testing manually.